pkg/daemon/daemon.go

```diff
@@ -149,19 +157,128 @@ func (d *daemon) GetIbSriovNetwork(networkID string) (string, *utils.IbSriovCniS
 	return networkName, ibCniSpec, nil
 }
 
+// Get Pod's IB network specified by name. Update netMap with Pod's networks.
+func getIbSriovPodNetwork(netName string, pw *PodWrapper, netMap map[types.UID][]*v1.NetworkSelectionElement) error {
+	var err error
+	networks, ok := netMap[pw.pod.UID]
+	if !ok {
+		networks, err = netAttUtils.ParsePodNetworkAnnotation(pw.pod)
+		if err != nil {
+			return fmt.Errorf("failed to read pod networkName annotations pod namespace %s name %s, with error: %v",
+				pw.pod.Namespace, pw.pod.Name, err)
+		}
+
+		netMap[pw.pod.UID] = networks
+	}
+	network, err := utils.GetPodNetwork(networks, netName)
+	if err != nil {
+		return fmt.Errorf("failed to get pod networkName spec %s with error: %v", netName, err)
+	}
+
+	pw.ibNetwork = network
+	pw.networks = networks
+	return nil
+}
+
+// Verify if GUID already exist for given network ID and allocates new one if not
+func (d *daemon) mapGUIDToPodNetwork(allocatedGUID, podNetworkID string, podUID types.UID) error {
+	if _, exist := d.guidPodNetworkMap[allocatedGUID]; exist {
```

Comment:
replate _ with podNetworkID to save additional access to map in the line below

or better yet just:
```
if allocatedPodNetworkID, _ := d.guidPodNetworkMap[allocatedGUID]; allocatedPodNetworkID != podNetworkID {
    return fmt.Errorf("failed to allocate......", allocatedGUID, allocatedPodNetworkID)
} 
```

Reply:
Second case wouldn't work. Lets just stick to replate.